src/bitcoin-cli.cpp

```diff
@@ -540,15 +540,15 @@ class NetinfoRequestHandler : public BaseRequestHandler
         // Report detailed peer connections list sorted by direction and minimum ping time.
         if (DetailsRequested() && !m_peers.empty()) {
             std::sort(m_peers.begin(), m_peers.end());
-            result += strprintf("<->   type   net  v  mping   ping send recv  txn  blk  hb %*s%*s%*s ",
-                                m_max_addr_processed_length, "addrp",
-                                m_max_addr_rate_limited_length, "addrl",
-                                m_max_age_length, "age");
+            result += tfm::format_raw("<->   type   net  v  mping   ping send recv  txn  blk  hb %*s%*s%*s ",
+                                      m_max_addr_processed_length, "addrp",
+                                      m_max_addr_rate_limited_length, "addrl",
+                                      m_max_age_length, "age");
             if (m_is_asmap_on) result += " asmap ";
-            result += strprintf("%*s %-*s%s\n", m_max_id_length, "id", IsAddressSelected() ? m_max_addr_length : 0, IsAddressSelected() ? "address" : "", IsVersionSelected() ? "version" : "");
+            result += tfm::format_raw("%*s %-*s%s\n", m_max_id_length, "id", IsAddressSelected() ? m_max_addr_length : 0, IsAddressSelected() ? "address" : "", IsVersionSelected() ? "version" : "");
             for (const Peer& peer : m_peers) {
                 std::string version{ToString(peer.version) + peer.sub_version};
-                result += strprintf(
+                result += tfm::format_raw(
```

Comment:
do we need to keep the `strprintf("%s%s"` on line 563?

Reply:
I suppose we don't, but that seems unrelated to this PR so going to leave that as is.

Reply:
Can you expand on that? We've changed other `strprintf` instance here to enable compile-time validation, how come this one didn't make the cut?

Reply:
Hmm, then I don't think I understand what you mean. Would you be able to provide a diff/example?

Reply:
My mistake, rereading it it's not clear what I was asking.
So I think we could simplify the nested formats by pulling up the internal pattern:
Having `%2s` in the main pattern and having a `%s%s` inside could be simplified to having `%s%s` at the parent and layout out the values in the parameters:
```patch
Index: src/bitcoin-cli.cpp
<+>UTF-8
===================================================================
diff --git a/src/bitcoin-cli.cpp b/src/bitcoin-cli.cpp
--- a/src/bitcoin-cli.cpp	(revision 2e33dbadcf69838c8ea591dccdc666f809c3134a)
+++ b/src/bitcoin-cli.cpp	(date 1727693490146)
@@ -549,7 +549,7 @@
             for (const Peer& peer : m_peers) {
                 std::string version{ToString(peer.version) + peer.sub_version};
                 result += tfm::format_raw(
-                    "%3s %6s %5s %2s%7s%7s%5s%5s%5s%5s  %2s %*s%*s%*s%*i %*s %-*s%s\n",
+                    "%3s %6s %5s %2s%7s%7s%5s%5s%5s%5s  %s%s %*s%*s%*s%*i %*s %-*s%s\n",
                     peer.is_outbound ? "out" : "in",
                     ConnectionTypeForNetinfo(peer.conn_type),
                     peer.network,
@@ -560,7 +560,8 @@
                     peer.last_recv ? ToString(time_now - peer.last_recv) : "",
                     peer.last_trxn ? ToString((time_now - peer.last_trxn) / 60) : peer.is_tx_relay ? "" : "*",
                     peer.last_blck ? ToString((time_now - peer.last_blck) / 60) : "",
-                    strprintf("%s%s", peer.is_bip152_hb_to ? "." : " ", peer.is_bip152_hb_from ? "*" : " "),
+                    peer.is_bip152_hb_to ? "." : " ",
+                    peer.is_bip152_hb_from ? "*" : " ",
                     m_max_addr_processed_length, // variable spacing
                     peer.addr_processed ? ToString(peer.addr_processed) : peer.is_addr_relay_enabled ? "" : ".",
                     m_max_addr_rate_limited_length, // variable spacing
Index: src/test/util_string_tests.cpp
<+>UTF-8
===================================================================
diff --git a/src/test/util_string_tests.cpp b/src/test/util_string_tests.cpp
--- a/src/test/util_string_tests.cpp	(revision 2e33dbadcf69838c8ea591dccdc666f809c3134a)
+++ b/src/test/util_string_tests.cpp	(date 1727693570815)
@@ -96,5 +96,10 @@
     BOOST_CHECK_EQUAL(oss.str(), R"(Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: "%.*f")");
 }
 
+BOOST_AUTO_TEST_CASE(TwoCharFormattingTest)
+{
+    BOOST_CHECK_EQUAL(tfm::format("%2s", strprintf("%s%s", ".", "*")),
+                      tfm::format("%s%s", ".", "*"));
+}
 
 BOOST_AUTO_TEST_SUITE_END()
```

(the dummy test was added just to make sure we're doing it correctly)

Reply:
Thanks for providing the diff. There are probably plenty of opportunities to clean up the bitcoin-cli string printing, but this line is not touched and it does not relate to the goal of this PR, so it seems best to me to leave this as is.

Reply:
Ok, thanks for checking